README.md

```diff
@@ -305,6 +306,30 @@ method takes one argument:
 
 2. `chain` must return a value of the same Chain
 
+### ChainRec
+
+A value that implements the ChainRec specification must also implement the Chain specification.
+
+1. `T.chainRec((done, next, value) => g(done(value)), i) `is equivalent to `g(i)`
```

Comment:
@rpominov type of `g` is ` a => T a`. maybe I have not clearly expressed that with words. it looks like  there is no word on `f/g`, for example in monad spec . should I remove the line about `g`? or what you think is best wording for that?

What you think on something like this?
> `g` returns value of type `T` which contains value of it's first argument type.

---

About the other laws we could state this

>`T.chainRec((next,done,v) => p(v) ? f(v).map(done) : g(v).map(next), i)` is equivalent to 
>```js
(function step(v) {
  return p(v) ? f(v) : g(v).chain(step)
}(i))
>```
>But `chainRec` **must use constant stack space**


And types would be:
```hs
p :: a -> bool
g :: a -> T b
f :: a -> T a
step :: a -> T b
```

And we could also include this in `laws` like this:

```js
const someNameHere = t => eq => p => f => g => x => {
    const a = t.chainRec((next,done,v) => p(v) ? f(v).map(done) : g(v).map(next), x);
    const b = (function step(v) { return p(v) ? f(v) : g(v).chain(step) }(x));
    return eq(a, b);
};
```
But this must be tested with small inputs as `step` can blow the stack :D

If we call first law `identity` then what the name should be for this one? /cc @SimonRichardson